worker/formatting_stack/background.clj

```diff
@@ -4,15 +4,23 @@
 
 (defonce workload (atom nil))
 
-(defonce runner
-  (future
-    (while (not (-> (Thread/currentThread) .isInterrupted))
-      (if-let [job @workload]
-        (when (compare-and-set! workload job nil)
-          (try
-            (job)
-            (catch Exception e
-              (-> e .printStackTrace))
-            (catch AssertionError e
-              (-> e .printStackTrace))))
-        (Thread/sleep 50)))))
+(defonce ^Thread
+  ^{:doc "The runner for 'background' execution. Can be stopped via the thread interruption mechanism."}
+  runner
+  (let [f (fn []
+            (while (not (-> (Thread/currentThread) .isInterrupted))
+              (if-let [job @workload]
+                (when (compare-and-set! workload job nil)
+                  (try
+                    (job)
+                    (catch Exception e
+                      (-> e .printStackTrace))
+                    (catch AssertionError e
+                      (-> e .printStackTrace))))
+                (Thread/sleep 50))))]
+    (-> f
+        Thread.
+        ;; Important - daemonize this thread, otherwise under certain conditions it can prevent the JVM from exiting.
+        ;; (We exercise this implicitly via `lein eastwood` in CI)
```

Comment:
As the comment hints, the thing being fixed here is that a non-daemonized thread can leave the JVM hanging forever.

I certainly have experieced that with f-s a bunch of times, it's very annoying.